9.tcga-classify/scripts/nbconverted/interpret-top-dae-tp53-model.py

```diff
@@ -0,0 +1,126 @@
+
+# coding: utf-8
+
+# ## Generating a Table of the BioBombe Interpreted Features in a High Scoring DAE model of TP53 inactivation
+# 
+# **Gregory Way, 2019**
+# 
+# I use the model previously identified that was used to predict TP53 inactivation.
+# I observe the BioBombe gene set enrichment scores for the features with high coefficients in this model.
+
+# In[1]:
+
+
+import os
+import sys
+import pandas as pd
+
+
+# ## Load the Top Model Identified Previously
+
+# In[2]:
+
+
+model_file = os.path.join("results", "top_dae_tp53_feature_for_followup.tsv")
+top_model_df = pd.read_table(model_file)
+top_model_df
+
+
+# In[3]:
+
+
+seed = top_model_df.seed.values[0]
+z_dim = top_model_df.z_dim.values[0]
+algorithm = top_model_df.algorithm.values[0].lower()
+
+
+# ## Load the BioBombe network projection results for Cancer Hallmarks
+
+# In[4]:
+
+
+file = os.path.join("..", "6.analyze-weights", "results", "tcga",
+                    "gph", "signal", "tcga_z_200_GpH__geneset_scores.tsv.gz")
+
+scores_df = (
+    pd
+    .read_table(file)
+    .query("seed == @seed")
+    .query("z == @z_dim")
+    .query("algorithm == @algorithm")
+)
+
+scores_df = (
+    scores_df
+    .assign(full_feature=scores_df.algorithm.astype(str) + "_" + scores_df.feature.astype(str),
+            abs_z_score=scores_df.z_score.abs())
+)
+scores_df.head()
+
+
+# ## Load Model Coefficients
+
+# In[5]:
+
+
+file = os.path.join("results", "mutation", "TP53", "TP53_coefficients.tsv.gz")
+
+top_n_features = 8
```

Comment:
Is there a reason this is 8 instead of 10?

Reply:
These are the features that are highlighted in the figure. I think it would be good to do 10 though. Will update!